src/app/clusters/device-energy-management-server/device-energy-management-server.h

```diff
@@ -160,25 +160,40 @@ class Delegate
 
     // ------------------------------------------------------------------
     // Get attribute methods
-    virtual ESATypeEnum GetESAType()                                                                       = 0;
-    virtual bool GetESACanGenerate()                                                                       = 0;
-    virtual ESAStateEnum GetESAState()                                                                     = 0;
-    virtual int64_t GetAbsMinPower()                                                                       = 0;
-    virtual int64_t GetAbsMaxPower()                                                                       = 0;
-    virtual DataModel::Nullable<Structs::PowerAdjustCapabilityStruct::Type> GetPowerAdjustmentCapability() = 0;
-    virtual DataModel::Nullable<Structs::ForecastStruct::Type> GetForecast()                               = 0;
-    virtual OptOutStateEnum GetOptOutState()                                                               = 0;
+    virtual ESATypeEnum GetESAType()                                                                         = 0;
+    virtual bool GetESACanGenerate()                                                                         = 0;
+    virtual ESAStateEnum GetESAState()                                                                       = 0;
+    virtual int64_t GetAbsMinPower()                                                                         = 0;
+    virtual int64_t GetAbsMaxPower()                                                                         = 0;
+    virtual OptOutStateEnum GetOptOutState()                                                                 = 0;
+
+    /**
+     * @brief Returns the current PowerAdjustCapability object
+     *
+     * The caller of GetPowerAdjustmentCapability() must not store a pointer/reference to the
+     * DataModel::Nullable<Structs::PowerAdjustCapabilityStruct::Type> return value. Structs::PowerAdjustCapabilityStruct
+     * contains a list of powerAdjustCapabilities (of type PowerAdjustStruct). The memory allocated for these
+     * is managed by the application and could change once the caller of GetPowerAdjustmentCapability has finished its
+     * processing.
+     *
+     * @return  The current PowerAdjustCapability object
+     */
+    virtual DataModel::Nullable<Structs::PowerAdjustCapabilityStruct::Type> & GetPowerAdjustmentCapability() = 0;
+
+    /**
+     * @brief Returns the current Forecast object
+     *
+     * The caller of GetForecast() must not store a pointer/reference to the Structs::ForecastStruct return value.
```

Comment:
Suggest a more explicit instruction to the writers of delegate applications:
```suggestion
 * The delegate MUST implement an API to provide a temporary reference to the 
     * Structs::ForecastStruct. The SDK (and other users) MUST not use 
     * the return value from GetForecast() after they have used it in the scope of the calling function.
     * The memory associated with the ForecastStruct is likely to change or be re-allocated, so would become invalid outside of the scope of this calling function.
```